### DISCUSSION TITLE
PyScript failure mode.

### DISCUSSION THREAD

**Discussion by ntoll** [pyscript/pyscript#2070] [Proposals]
While at PyCon, because of flaky conference WiFi, sometimes packages or other assets failed to load and PyScript was left in a broken state. This broken state was not communicated to the user in a meaningful manner. 

It was suggested that we should provide a helpful message to the user so they can at least understand PyScript has not been able to start because of some failure they can further investigate.

We need to figure out how this might look and what meaningful and actionable information we show to the user should PyScript fail to start. We also need to clearly define what failure conditions we report (e.g. a timeout when trying to download a package).

**Comment by WebReflection** [comment]
we don't have hard-coded timeouts for anything ... maybe we should, and when this happens, we use the error (that might be disabled by users). Can I ask what exactly didn't work and from where? handling all possible 3rd party fetch operations, including those done by micropip or mip, is not straight forward but the devtools network panel should show errors, if any, in there ... if I could better understand what was the specific condition, I might think of a workaround but without a Service Worker it's very hard to interfere or intercept all possible network requests ... I mean, it could be a `link` CSS / style error on the HTML and we don't get to understand that failed from our code, as example, so you'll see "broken UI" that has nothing to do with PyScript ... if that makes sense.
